zerver/views/users.py

```diff
@@ -114,6 +114,10 @@ def deactivate_user_backend(
     deactivation_notification_comment: Optional[str] = REQ(
         str_validator=check_capped_string(max_length=2000), default=None
     ),
+    spammer: Optional[bool] = REQ(default=None, json_validator=check_bool),
+    message_delete_action: Optional[int] = REQ(
+        default=None, json_validator=check_int_in(Message.DEACTIVATED_USER_MESSAGE_DELETE_ACTION)
+    ),
```

Comment:
So after some discussion with @alya, I think it's probably better replace the current dropdown design with 3 checkboxes:
* Delete all private messages
* Delete all public stream messages
* Delete all private stream messages

With the default being all checked if you selected "Mark as spammer". 

The extra flexibility seems useful and it's probably about as simple from an API perpsective.

We can call the checkboxes `delete_private_messages`, `delete_public_stream_messages`, `delete_private_stream_messages`.